### PR TITLE
fix(framework): Added split payment and Split Refund Support

### PR DESCRIPTION
**Pull request by Shubhodip900**
## Description

Added Split Payment and Split Refund support in proto and domain.
Connector support of Adyen and Stripe.
### Additional Changes
- [ ] This PR modifies the API contract
- [ ] This PR modifies application configuration/environment variables

## How did you test it?

Tested for Stripe Connector:-

AUthorize Request:-
```json
{
    "amount": 1000,
    "currency": "USD",
    "confirm": true,
    "capture_method": "automatic",
    "customer_id": "cust_no_pm",
    "authentication_type": "no_three_ds",
      "email": "guest@example.com",
    "payment_method": "card",
    "payment_method_type": "debit",
    "payment_method_data": {
        "card": {
            "card_number": "4242424242424242", 
            "card_exp_month": "03",
            "card_exp_year": "30",
            "card_holder_name": "joseph Doe",
            "card_cvc": "737"
        }
    },
    "billing": {
        "address": {
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "city": "San Fransico",
            "state": "California",
            "zip": "94122",
            "country": "US",
            "first_name": "joseph",
            "last_name": "Doe"
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        }
    },
    "split_payments": {
        "stripe_split_payment": {
            "charge_type": "destination",
            "application_fees": 100,
            "transfer_account_id": "********"
        }
    },
    "browser_info": {
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "language": "nl-NL",
        "color_depth": 24,
        "screen_height": 723,
        "screen_width": 1536,
        "time_zone": 0,
        "java_enabled": true,
        "java_script_enabled": true,
        "ip_address": "13.232.74.226"
    },
    "customer_acceptance": {
        "acceptance_type": "online",
        "accepted_at": "1963-05-03T04:07:52.723Z",
        "online": {
            "ip_address": "127.0.0.1",
            "user_agent": "amet irure esse"
        }
    },
    "setup_future_usage": "on_session"

    
}
```

Response:-
```json
{
    "payment_id": "pay_dKC0qZ4MI2uX6NWdcP3x",
    "merchant_id": "merchant_1781696080",
    "status": "succeeded",
    "amount": 1000,
    "net_amount": 1000,
    "shipping_cost": null,
    "amount_capturable": 0,
    "amount_received": 1000,
    "processor_merchant_id": "merchant_1781696080",
    "initiator": null,
    "sdk_authorization": "cHJvZmlsZV9pZD1wcm9fNjdBUlZYeE5sRFdKcE9rNmFCanIscHVibGlzaGFibGVfa2V5PXBrX2Rldl8wZjM3NTVmYWY1OWI0YTI4OTNlZGUzYzQ3NWI5NjgyOSxjbGllbnRfc2VjcmV0PXBheV9kS0MwcVo0TUkydVg2TldkY1AzeF9zZWNyZXRfbVQ5UEE3cndNYThIWVhyOWpSYUosY3VzdG9tZXJfaWQ9Y3VzdF9ub19wbSxjbGllbnRfc2Vzc2lvbl9pZD1jbGllbnRfc2Vzc19DV1gxcnhDQ3pCR3JoZ1RaOE1IVyxwYXltZW50X2lkPXBheV9kS0MwcVo0TUkydVg2TldkY1AzeA==",
    "connector": "stripe",
    "state_metadata": null,
    "client_secret": "pay_dKC0qZ4MI2uX6NWdcP3x_secret_mT9PA7rwMa8HYXr9jRaJ",
    "created": "2026-06-17T11:36:17.248Z",
    "modified_at": "2026-06-17T11:36:20.919Z",
    "connector_customer_id": "cus_UijDwVn3eHhM7c",
    "currency": "USD",
    "customer_id": "cust_no_pm",
    "customer": {
        "id": "cust_no_pm",
        "name": null,
        "email": "guest@example.com",
        "phone": null,
        "phone_country_code": null,
        "customer_document_details": null
    },
    "description": null,
    "refunds": null,
    "disputes": null,
    "mandate_id": null,
    "mandate_data": null,
    "setup_future_usage": "on_session",
    "off_session": null,
    "capture_on": null,
    "capture_method": "automatic",
    "payment_method": "card",
    "payment_method_data": {
        "card": {
            "last4": "4242",
            "card_type": null,
            "card_network": null,
            "card_issuer": null,
            "card_issuing_country": null,
            "card_isin": "424242",
            "card_extended_bin": null,
            "card_exp_month": "03",
            "card_exp_year": "30",
            "card_holder_name": "joseph Doe",
            "payment_checks": {
                "cvc_check": "pass",
                "address_line1_check": "pass",
                "address_postal_code_check": null
            },
            "authentication_data": null,
            "auth_code": null
        },
        "billing": null
    },
    "payment_token": null,
    "shipping": null,
    "billing": {
        "address": {
            "city": "San Fransico",
            "country": "US",
            "line1": "1467",
            "line2": "Harrison Street",
            "line3": "Harrison Street",
            "zip": "94122",
            "state": "California",
            "first_name": "joseph",
            "last_name": "Doe",
            "origin_zip": null
        },
        "phone": {
            "number": "8056594427",
            "country_code": "+91"
        },
        "email": null
    },
    "order_details": null,
    "email": "guest@example.com",
    "name": null,
    "phone": null,
    "return_url": null,
    "authentication_type": "no_three_ds",
    "statement_descriptor_name": null,
    "statement_descriptor_suffix": null,
    "next_action": null,
    "cancellation_reason": null,
    "error_code": null,
    "error_message": null,
    "unified_code": null,
    "unified_message": null,
    "error_details": null,
    "payment_experience": null,
    "payment_method_type": "debit",
    "connector_label": null,
    "business_country": null,
    "business_label": "default",
    "business_sub_label": null,
    "allowed_payment_method_types": null,
    "manual_retry_allowed": null,
    "connector_transaction_id": "pi_3TjHlWD5R7gDAGff06e550cX",
    "frm_message": null,
    "metadata": null,
    "connector_metadata": null,
    "connector_response_metadata": null,
    "feature_metadata": {
        "redirect_response": null,
        "search_tags": null,
        "apple_pay_recurring_details": null,
        "pix_additional_details": null,
        "boleto_additional_details": null,
        "pix_automatico_additional_details": null,
        "finix_additional_details": null
    },
    "reference_id": "pi_3TjHlWD5R7gDAGff06e550cX",
    "payment_link": null,
    "profile_id": "pro_67ARVXxNlDWJpOk6aBjr",
    "surcharge_details": null,
    "attempt_count": 1,
    "merchant_decision": null,
    "merchant_connector_id": "mca_803AgYXPX1elagg2sHZp",
    "incremental_authorization_allowed": false,
    "authorization_count": null,
    "incremental_authorizations": null,
    "external_authentication_details": null,
    "external_3ds_authentication_attempted": false,
    "expires_on": "2026-06-17T11:51:17.248Z",
    "fingerprint": null,
    "browser_info": {
        "language": "nl-NL",
        "time_zone": 0,
        "ip_address": "13.232.74.226",
        "user_agent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/70.0.3538.110 Safari/537.36",
        "color_depth": 24,
        "java_enabled": true,
        "screen_width": 1536,
        "accept_header": "text/html,application/xhtml+xml,application/xml;q=0.9,image/webp,image/apng,*/*;q=0.8",
        "screen_height": 723,
        "java_script_enabled": true
    },
    "payment_channel": null,
    "payment_method_id": null,
    "network_transaction_id": "104557651805572",
    "network_transaction_link_id": null,
    "payment_method_status": null,
    "updated": "2026-06-17T11:36:20.919Z",
    "split_payments": {
        "stripe_split_payment": {
            "charge_id": "ch_3TjHlWD5R7gDAGff0YV5hmZ4",
            "charge_type": "destination",
            "application_fees": 100,
            "transfer_account_id": "*********",
            "on_behalf_of": null
        }
    },
    "frm_metadata": null,
    "extended_authorization_applied": false,
    "extended_authorization_last_applied_at": null,
    "request_extended_authorization": null,
    "capture_before": null,
    "merchant_order_reference_id": null,
    "order_tax_amount": null,
    "connector_mandate_id": null,
    "card_discovery": "manual",
    "force_3ds_challenge": false,
    "force_3ds_challenge_trigger": false,
    "issuer_error_code": null,
    "issuer_error_message": null,
    "is_iframe_redirection_enabled": null,
    "whole_connector_response": null,
    "enable_partial_authorization": null,
    "enable_overcapture": null,
    "is_overcapture_enabled": false,
    "network_details": null,
    "is_stored_credential": null,
    "mit_category": null,
    "billing_descriptor": null,
    "tokenization": null,
    "partner_merchant_identifier_details": null,
    "payment_method_tokenization_details": null,
    "installment_options": null,
    "installment_data": null,
    "sender_payment_instrument_id": null
}
```

### CODE REVIEW

File: crates/integrations/connector-integration/src/connectors/stripe/transformers.rs [hunk 1]
```diff
@@ -4347,6 +4344,10 @@ impl<F> TryFrom<&RouterDataV2<F, RefundFlowData, RefundsData, RefundsResponseDat
                         },
                     })
                 }
+                _ => Err(IntegrationError::MissingRequiredField {
+                    field_name: "stripe_split_refund",
+                    context: Default::default(),
```
Comment (gopikrishna000) [inline crates/integrations/connector-integration/src/connectors/stripe/transformers.rs]: can we have message

**Comment by XyneSpaces** [comment]
## Code Review: Split Payment and Split Refund Support

### 🔍 Findings

#### 1. **Enum Definition Review** ✅
**File:** `crates/common/common_enums/src/enums.rs`

New `AdyenSplitType` enum properly derives required traits:
- `Clone, Debug, Eq, PartialEq` for standard behavior
- `serde::Deserialize, serde::Serialize` for wire compatibility
- `strum::Display, strum::EnumString` for string conversions
- `ToSchema` for API documentation

Serde and strum both use `PascalCase` rename - consistent with convention.

#### 2. **Documentation Quality** ⚠️
Several variant doc comments have leading double-spaces (e.g., `"  The transaction fees..."`, `"  The fees paid..."`). These should be normalized to single space for consistency.

#### 3. **Split Payment Flow Implementation Check Required**
Per scenario classification (`core-flow-framework` + `connector-flow-addition`):

- [ ] Verify new split fields in proto/gRPC contracts align with Hyperswitch backend changes
- [ ] Confirm `ConnectorIntegrationV2` implementations for Adyen and Stripe handle split fields
- [ ] Check amount calculations preserve split allocation math (no floating-point rounding issues)

#### 4. **Wire Compatibility**
Adding new enum variants to `common_enums` has blast radius across workspace. Verify:
- All pattern matches on `AdyenSplitType` (when added elsewhere) are exhaustive
- No `_ =>` catch-all that would silently drop new variants

#### 5. **Connector-Specific Concerns**
For Adyen split payments:
- Ensure `BalanceAccount` splits validate account existence before transaction
- Confirm split amounts sum to total payment amount (risk of mismatch)

For Stripe Connect refunds:
- Verify `split_refunds` routing follows Stripe's reverse transfer semantics

### Scenario Classification
- **Primary:** `core-flow-framework` (shared enum changes)
- **Secondary:** `connector-flow-addition` (Adyen/Stripe split implementations)
- **Tertiary:** `proto-api-contract` (if proto fields added)

### Recommendations
1. Fix leading double-spaces in doc comments
2. Ensure split amount validation at connector level (sum checks)
3. Add test specs covering split payment scenarios for both connectors

**Comment by gopikrishna000** [comment]
Proto review — split payment/refund types look good overall; field tags, `oneof` per connector, and `*_UNSPECIFIED = 0` zero values are all consistent with the file. A few things worth tightening:

**1. `optional` vs implicit-required is inconsistent on the split scalars.** In `AdyenSplitItem`, `amount`/`account`/`description` are `optional` but `split_type`/`reference` are not; `StripeSplitRefund.charge_id`/`transfer_account_id` and `StripeSplitPaymentRequest.transfer_account_id` are also implicit-required. The rest of the file wraps scalars in `optional`. Can we make these deliberate — `optional` for anything not strictly mandatory?

**2. Leaf message naming is mixed:** `StripeSplitPaymentRequest` / `StripeSplitRefund` / `AdyenSplitData`. These are nested config payloads, not RPC requests (the enclosing `SplitPaymentsRequest`/`SplitRefundsRequest` own that role). Suggest dropping the suffix to `StripeSplitPayment` so the Stripe pair is symmetric (`StripeSplitPayment` / `StripeSplitRefund`), and keep `AdyenSplitData` as-is — the neutral noun is correct since one type serves both flows.

**3. `ChargeRefundsOptions.option`** — the field name `option` shadows the proto `option` keyword and can read awkwardly in codegen; `charge_option` is clearer.

**4. Xendit gap:** the `oneof`s cover Stripe + Adyen only, while the domain enum also has Xendit variants (mapped to `None` consumer-side). Fine for this PR's stated scope — just flagging so Xendit-via-UCS isn't assumed working downstream.

**Review by XyneSpaces — COMMENTED** [review]
## Review Summary

**Classification:** connector-flow-addition · core-flow-framework · connector-payment-method-addition
**Risk:** High (payment-critical paths affected)

**Verdict:** :arrows_counterclockwise: Request Changes

:rotating_light: 1 critical · :warning: 2 warnings · :bulb: 2 suggestions

---

### :rotating_light: Critical: Missing `None` arm in Stripe split payment extraction

**File:** `crates/integrations/connector-integration/src/connectors/stripe.rs`

The new `.and_then()` extraction for `StripeSplitPayment` returns `None` for any non-Stripe split payment variant. This silently drops Adyen split payments when routed through the same code path, potentially causing shadow-validation diffs.

**Fix:** Explicitly handle the `None` case with a proper error or ensure this code path is never reached with non-Stripe variants. If this is connector-specific code, consider an `unreachable!()` with explanatory comment for other variants.

---

### :warning: Warning: Hard-coded default values in Adyen split request

**File:** `crates/integrations/connector-integration/src/connectors/adyen/transformers.rs`

The `get_adyen_split_request()` function silently falls back to `adyen_metadata.store.clone()` when `split_payments` is `None` or not an Adyen variant. This may cause incorrect split routing when the merchant intended no splits.

**Fix:** Verify that falling back to metadata store is the intended behavior. If no split is requested, the function should return `(None, None)` to explicitly indicate no splitting.

---

### :warning: Warning: Amount unit conversion not verified

**File:** `crates/integrations/connector-integration/src/connectors/adyen/transformers.rs`

The `AdyenSplitItem.amount` is `Option<MinorUnit>` but the PR doesn't show the conversion logic. Verify that:
1. The amount is correctly converted from the router's currency unit
2. The conversion is consistent with Adyen's expected format
3. Test coverage includes edge cases (zero amounts, large amounts, currency boundaries)

---

### :bulb: Suggestion: Add exhaustive match test for AdyenSplitType

**File:** `crates/common/common_enums/src/enums.rs`

With 11 new split types, consider adding a test that verifies all variants serialize to the correct Adyen API strings. This prevents typos in the snake_case naming from causing integration failures.

---

### :bulb: Suggestion: Document split payment precedence

**File:** `crates/types-traits/domain_types/src/connector_types.rs`

Add documentation clarifying which split payment type takes precedence when multiple are present (e.g., if both Stripe and Adyen splits are somehow populated).

---

**Missing companion verification:**
- [ ] Confirm `data/field_probe/manifest.json` updated for new Adyen split fields
- [ ] Verify test specs in `crates/internal/ucs-connector-tests/` cover split payment scenarios

File: crates/types-traits/grpc-api-types/proto/payment.proto [hunk 2]
```diff
@@ -2744,6 +2750,9 @@ message PaymentServiceRefundRequest {
 
   // Optional typed payment method (e.g. wallet number for direct-wallet refunds).
   optional PaymentMethod payment_method = 19;
+
+  // Split refund data for platform fee reversals and transfers
```
Comment (XyneSpaces) [inline crates/types-traits/grpc-api-types/proto/payment.proto:2878]: :warning: **Unspecified variant handling lacks error context**

The `AdyenSplitType` enum uses `UNSPECIFIED = 0` as the default. When `foreign_try_from` encounters this variant, it returns an `InvalidDataFormat` error, which is good. However, consider whether the protobuf definition should mark this as `reserved` or document that `UNSPECIFIED` is an invalid state to prevent accidental usage.

**Fix**: Add a comment in the proto explaining that `UNSPECIFIED` is not a valid operational value and must not be used in requests.

File: crates/types-traits/domain_types/src/types.rs [hunk 3]
```diff
@@ -3030,6 +3031,7 @@ impl From<grpc_payment_types::PaymentServiceAuthorizeRequest> for AuthorizationR
             payment_method_token: None,
             merchant_request_id: req.merchant_request_id,
             connector_order_id: req.connector_order_id,
+            split_payments: req.split_payments,
         }
     }
 }
```
Comment (XyneSpaces) [inline crates/types-traits/domain_types/src/types.rs:3040]: :rotating_light: **Silent error swallowing with `unwrap_or_default()`**

The code uses `unwrap_or_default()` when converting proto enum values:
```rust
grpc_api_types::payments::AdyenSplitType::try_from(value.split_type)
    .unwrap_or_default()
```

This silently converts any unknown/invalid split type values to `Unspecified`, which then fails later with a generic error. This makes debugging difficult when new split types are added to the proto but not yet supported in the domain code.

**Fix**: Propagate the conversion error explicitly instead of using `unwrap_or_default()`. If the proto contains an unknown variant, the error should indicate which variant was received.

**Review by XyneSpaces — COMMENTED** [review]
Review findings for split payment and split refund support PR.

File: crates/types-traits/domain_types/src/types.rs [hunk 4]
```diff
@@ -3685,7 +3688,10 @@ impl<
...
             merchant_config_currency: Some(merchant_config_currency),
             all_keys_required: None, // Field not available in new proto structure
-            split_payments: None,
+            split_payments: value
+                .split_payments
+                .map(connector_types::SplitPaymentsRequest::foreign_try_from)
+                .transpose()?,
             enable_overcapture: None,
             setup_mandate_details: value
                 .setup_mandate_details
```
Comment (XyneSpaces) [inline crates/types-traits/domain_types/src/types.rs:3883]: :warning: **Same `unwrap_or_default()` issue for `PaymentChargeType`**

```rust
grpc_api_types::payments::PaymentChargeType::try_from(value.charge_type)
    .unwrap_or_default()
```

This silently defaults to `Unspecified` on conversion failure, losing information about what invalid value was received.

**Fix**: Handle the error case explicitly and propagate it with context about the invalid charge type value.

File: crates/types-traits/domain_types/src/connector_types.rs [hunk 5]
```diff
@@ -3827,6 +3829,9 @@ pub struct StripeSplitPaymentRequest {
 
     /// Identifier for the reseller's account where the funds were transferred
     pub transfer_account_id: String,
+
+    /// The Stripe account ID that these funds are intended for
+    pub on_behalf_of: Option<String>,
```
Comment (XyneSpaces) [inline crates/types-traits/domain_types/src/connector_types.rs:3839]: :bulb: **`deny_unknown_fields` on `AdyenSplitData`**

The `#[serde(deny_unknown_fields)]` attribute on `AdyenSplitData` and `AdyenSplitItem` is good for catching typos in JSON configs. However, this could cause deserialization failures if Adyen adds new optional fields to their split API response that we don't yet handle.

**Fix**: Consider whether this should be relaxed to `#[serde(default)]` on optional new fields, or document that this struct must be updated whenever Adyen's split API changes.

File: sdk/javascript/src/payments/_generated_grpc_client.ts [hunk 6]
```diff
@@ -380,9 +380,9 @@ const _MSG_FIELD_TYPES: Record<string, Record<string, string>> = {
   ApplePayPaymentRequest: { "total": "AmountInfo" },
```
Comment (XyneSpaces) [inline sdk/javascript/src/payments/_generated_grpc_client.ts:385]: :bulb: **SDK multi-language parity check**

The JavaScript SDK has been updated with the new `SplitPaymentsRequest` and `SplitRefundsRequest` field mappings. Verify that the Python SDK (`sdk/python/src/payments/`) and any other language SDKs are also regenerated with these proto changes to maintain API parity across clients.

**Fix**: If Python/other SDKs haven't been regenerated, run the SDK generation Makefile targets and include those changes in this PR.

File: crates/integrations/connector-integration/src/connectors/stripe/transformers.rs [hunk 7]
```diff
@@ -2087,23 +2087,23 @@ impl<T: PaymentMethodDataTypes + Debug + Sync + Send + 'static + Serialize>
...
-                    common_enums::StripeChargeType::Destination => Some(IntentCharges {
-                        application_fee_amount: stripe_split_payment.application_fees,
-                        destination_account_id: Some(Secret::new(
-                            stripe_split_payment.transfer_account_id.clone(),
-                        )),
-                    }),
-                },
-            },
-            None => None,
+            Some(SplitPaymentsRequest::StripeSplitPayment(stripe_split_payment)) => {
```
Comment (XyneSpaces) [inline crates/integrations/connector-integration/src/connectors/stripe/transformers.rs]: :warning: **Exhaustive match on `SplitPaymentsRequest` variants**

The match on `split_payments` now handles:
- `Some(SplitPaymentsRequest::StripeSplitPayment(...))`
- `Some(SplitPaymentsRequest::AdyenSplitPayment(_)) | None`

This is correct but fragile—if a new split payment variant is added to the enum, this match will silently fall through to the `None` case.

**Fix**: Consider using a catch-all pattern that explicitly warns:
```rust
Some(other) => {
    // Log warning about unsupported split payment type for Stripe
    None
}
```

File: crates/integrations/connector-integration/src/connectors/adyen/transformers.rs [hunk 8]
```diff
@@ -25,7 +25,7 @@ use domain_types::{
         PaymentVoidData, PaymentsAuthorizeData, PaymentsCancelPostCaptureData, PaymentsCaptureData,
         PaymentsIncrementalAuthorizationData, PaymentsResponseData, PaymentsSyncData,
         RefundFlowData, RefundsData, RefundsResponseData, RepeatPaymentData, ResponseId,
-        SetupMandateRequestData, SubmitEvidenceData,
+        SetupMandateRequestData, SplitPaymentsRequest, SubmitEvidenceData,
     },
```
Comment (XyneSpaces) [inline crates/integrations/connector-integration/src/connectors/adyen/transformers.rs:29]: :bulb: **Refactored `get_adyen_split_request` signature change**

The function signature changed from:
```rust
fn get_adyen_split_request(metadata, adyen_store, currency)
```
to:
```rust
fn get_adyen_split_request(split_request, currency)
```

This is cleaner as it separates the extraction of `AdyenSplitData` from the request parsing. However, ensure all call sites (including any in tests or other connectors) are updated accordingly.

**Fix**: Verify no other call sites exist outside the changed files by grepping for `get_adyen_split_request`.

File: crates/grpc-server/grpc-server/tests/authorizedotnet_payment_flows_test.rs [hunk 9]
```diff
@@ -410,7 +410,7 @@ fn create_payment_get_request(transaction_id: &str) -> PaymentServiceGetRequest
         connector_order_reference_id: None,
```
Comment (XyneSpaces) [inline crates/grpc-server/grpc-server/tests/authorizedotnet_payment_flows_test.rs:410]: :mag: **Test fixture update pattern**

The test fixtures are being updated to include `split_payments: None` and `split_refunds: None`. This is necessary for compilation, but consider adding a test that exercises the split payment flow for at least one connector to ensure end-to-end functionality.

**Fix**: Add a test case in `stripe_payment_flows_test.rs` that exercises the split payment path with non-None values.

**Comment by XyneSpaces** [comment]
## Review Summary

This PR adds split payment and split refund support for Stripe and Adyen connectors.

### Key Findings

| Severity | Count | Category |
|----------|-------|----------|
| :rotating_light: Critical | 1 | Silent error swallowing with `unwrap_or_default()` on proto enum conversions |
| :warning: Warning | 3 | Unspecified variant handling, exhaustive match fragility, SDK parity |
| :bulb: Suggestion | 4 | Error propagation improvements, test coverage, documentation |

### Critical Issues

1. **`unwrap_or_default()` on proto enum conversions** - The code silently defaults to `Unspecified` when receiving unknown enum variants from the proto layer, making debugging difficult when new split types are added.

### Recommendations

1. Propagate proto enum conversion errors explicitly instead of using `unwrap_or_default()`
2. Add test coverage for split payment flows with actual split data (not just `None`)
3. Verify SDK parity across Python and other language clients
4. Document the `UNSPECIFIED` proto variant as invalid for operational use

### Scenarios Covered
- ✅ `proto-api-contract` - Proto schema changes reviewed
- ✅ `core-flow-framework` - Domain type changes reviewed  
- ✅ `connector-flow-addition` - Stripe/Adyen transformer changes reviewed
- ✅ `sdk-ffi-codegen` - Generated SDK changes reviewed
- ✅ `server-composite` - Composite service changes reviewed

**Comment by Shubhodip900** [comment]
After evaluating the different approaches, I believe keeping separate `StripeSplitPayment` and `StripeSplitRefund` types within their respective request flows is the cleanest design.

The alternative of introducing a common `StripeSplitData` wrapper containing both payment and refund variants has a few drawbacks:

1. It exposes refund-related structures in the payment flow (and vice versa), which does not accurately model the business operation being performed.
2. It allows semantically invalid states at the schema level, requiring additional runtime validation to reject them.
3. It makes the API less intuitive for consumers, as payment requests should only contain payment-related data and refund requests should only contain refund-related data.

Another alternative is introducing nested protobuf messages such as:

```proto
StripeSplitData.StripeSplitPayment
StripeSplitData.StripeSplitRefund
```

primarily for namespacing purposes. However, this also has some disadvantages:

1. It increases the verbosity of generated code (for example, `stripe_split_data::StripeSplitPayment` and `stripe_split_data::StripeSplitRefund` in Rust).
2. It adds an additional layer of nesting without providing any functional benefit.
3. The namespacing benefit is relatively small since the top-level message names are already sufficiently descriptive.
4. It can make the generated types and imports slightly harder to work with across the codebase.

Keeping the current separation provides stronger type safety, clearer API boundaries, and better aligns with the operation-oriented design followed across Hyperswitch. The small amount of duplication is acceptable here because `StripeSplitPayment` and `StripeSplitRefund` represent distinct business capabilities, even if some connectors (such as Adyen) happen to reuse the same underlying structure for both.

**Review by AmitsinghTanwar007 — DISMISSED** [review]

**Review by JeevaRamu0104 — DISMISSED** [review]

File: crates/types-traits/grpc-api-types/proto/composite_payment.proto [hunk 10]
```diff
@@ -177,6 +180,9 @@ message CompositeGetRequest {
   // Payment method type, supplied by the caller so connectors that need it on
   // sync (e.g. Adyen, to build connector_response) can populate it.
   optional PaymentMethodType payment_method_type = 18;
+
+  // Split payment configuration (platform fees, transfers, etc.)
+  optional SplitPaymentsRequest split_payments = 19;
```
Comment (gopikrishna000) [inline crates/types-traits/grpc-api-types/proto/composite_payment.proto]: can we have SplitPaymentsDetails instead of SplitPaymentsRequest

File: crates/types-traits/grpc-api-types/proto/payment.proto [hunk 11]
```diff
@@ -5099,3 +5119,100 @@ message PaymentServiceProxySetupRecurringRequest {
   optional string order_category = 16; // Category of the order/service
   optional string description = 17; // Description of the mandate setup
 }
+
+// ============================================================================
+// SPLIT REFUND TYPES
+// ============================================================================
+
+// Request for split refunds (platform fee reversals, transfers, etc.)
+message SplitRefundsRequest {
```
Comment (gopikrishna000) [inline crates/types-traits/grpc-api-types/proto/payment.proto]: SplitRefundsDetails

File: crates/types-traits/grpc-api-types/proto/payment.proto [hunk 12]
```diff
@@ -5099,3 +5119,100 @@ message PaymentServiceProxySetupRecurringRequest {
...
+  oneof split_refund_type {
+    StripeSplitRefund stripe_split_refund = 1;
```
Comment (gopikrishna000) [inline crates/types-traits/grpc-api-types/proto/payment.proto]: StripeSplitRefundData

**Review by gopikrishna000 — APPROVED** [review]

**Review by AmitsinghTanwar007 — APPROVED** [review]

**Review by JeevaRamu0104 — APPROVED** [review]
